octopoes/octopoes/core/service.py

```diff
@@ -162,7 +162,7 @@ def get_ooi_tree(
 
     def _delete_ooi(self, reference: Reference, valid_time: datetime) -> None:
         referencing_origins = self.origin_repository.list_origins(valid_time, result=reference)
-        if not referencing_origins:
+        if not [ro for ro in referencing_origins if ro.origin_type != OriginType.AFFIRMATION]:
```

Comment:
Nice find. I want to suggest the following to improve this even more:
```suggestion
        if not any(origin for origin in referencing_origins if origin.origin_type != OriginType.AFFIRMATION):
```
- The variable is more descriptive
- Simplified the condition; using this generator expression with `any` would be more Pythonic and efficient than creating a full list to check its emptiness